src/pages/Admin/BackgroundSelect/BackgroundSelect.tsx

```diff
@@ -48,45 +47,60 @@ export const BackgroundSelect: React.FC<BackgroundSelectProps> = ({
 
   const hasBackgrounds = !!mapBackgrounds.length && !isLoadingBackgrounds;
 
+  const renderBackground = useCallback(
+    (mapBackground: string, index: number) => (
+      <button
+        className="BackgroundSelect__map"
+        disabled={isUploading}
+        style={{ backgroundImage: `url(${mapBackground})` }}
+        key={index}
+        onClick={() => uploadMapBackground(mapBackground)}
+      />
+    ),
+    [isUploading, uploadMapBackground]
+  );
+
   const renderMapBackgrounds = useMemo(
-    () =>
-      mapBackgrounds.map((mapBackground, index) => (
-        <button
-          className="BackgroundSelect__map"
-          disabled={isUploading}
-          style={{ backgroundImage: `url(${mapBackground})` }}
-          key={index}
-          onClick={() => uploadMapBackground(mapBackground)}
-        />
-      )),
-    [isUploading, mapBackgrounds, uploadMapBackground]
+    () => mapBackgrounds.map(renderBackground),
+    [mapBackgrounds, renderBackground]
+  );
+
+  const renderDefaultBackgrounds = useMemo(
+    () => DEFAULT_BACKGROUNDS.map(renderBackground),
+    [renderBackground]
   );
 
   return (
     <div className="BackgroundSelect">
-      {!mapBackground && (
-        <>
-          <FileButton
-            disabled={isUploading}
-            title="Import a map background"
-            description="Recommended size: 2000px / 1200px"
-            onChange={uploadMapBackground}
-          />
-
-          <h3 className="BackgroundSelect__maps-header">
-            Or select one of our map backgrounds
-          </h3>
-          {isLoadingBackgrounds && <div>Loading maps...</div>}
-
-          <div className="BackgroundSelect__map-grid">
-            {hasBackgrounds && renderMapBackgrounds}
-          </div>
-        </>
-      )}
-
-      {mapBackground && (
-        <img width="100%" src={mapBackground} alt="Venue map background" />
-      )}
+      <>
+        <FileButton
+          disabled={isUploading}
+          title="Import a map background"
+          description="Recommended size: 2000px / 1200px"
+          onChange={uploadMapBackground}
+        />
+
+        <h3 className="BackgroundSelect__maps-header">
+          Or select one of our map backgrounds
+        </h3>
+
+        <div className="BackgroundSelect__map-grid">
+          {renderDefaultBackgrounds}
+          {hasBackgrounds && renderMapBackgrounds}
+        </div>
+
+        {isLoadingBackgrounds && <div>Loading maps...</div>}
+
+        {error && (
+          <>
+            <div>
+              The preset map backgrounds could not be fetched. Please, refresh
+              the page or upload a custom map background.
```

Comment:
This whole block can safely be moved to the parent component and keep concerns separate (the one who generates the error, shows it to the user).

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/2327/commits/a09cf6d18230c0649548810acf72d368b9735aa4